lab_3/main.py

```diff
@@ -222,7 +355,27 @@ def save(self, name: str) -> int:
         :param name: name of the json file with .json format
         :return: 0 if profile saves, 1 if any errors occurred
         """
-        pass
+        if not isinstance(name, str):
+            return 1
+        profile_as_dict = {}
+        freq_dict = {}
+
+        for trie in self.tries:
+            for key, value in trie.n_gram_frequencies.items():
+                string_for_file = ''
+                # (1, 2): 1;, where key = n-gram, value = how many times it appears
+                for element in key:
+                    string_for_file += self.storage.get_letter_by_id(element)
+                freq_dict[string_for_file] = value
+
+        profile_as_dict['freq'] = freq_dict
+        profile_as_dict['n_words'] = self.n_words
+        profile_as_dict['name'] = self.language
+        with open(name, 'w', encoding="UTF-8") as file:
+            # added parameter "encoding" because of code style
+            json_string = json.dumps(profile_as_dict)
+            file.write(json_string)
```

Comment:
There is a `.dump` method in the `json` package that combines these lines.

Reply:
Ok, I fixed this